docs/howto/freeze_a_ct_log.md

```diff
@@ -0,0 +1,133 @@
+# How To Freeze a CT Log
+
+## Prerequisites
+
+Some of the tools and metrics that will be used were only added recently.
+Before starting ensure that your server is upgraded to a release built
+from code corresponding to at least the commit tagged `v1.0.8`.
+
+The `log_signer` process(es) must be exporting metrics. Check that their
+`--http_endpoint` flag is set to an appropriate value. If it's empty then
+update the configuration and restart them before proceeding.
+
+## Preparation
+
+### Find the Log ID
+
+Obtain the ID of the tree that is backing the log that is to be frozen. This
+can be found in the CTFE config file. Locate the section of the config
+file that matches the log to be frozen and pull out the value of `log_id`.
+For example with the following config the `log_id` to use would be `987654321`.
+
+```
+config {
+	log_id: 987654321
+	prefix: "the_name_of_the_log"
+	roots_pem_file: "... roots file name ...."
+	public_key: {
+		der: ".... bytes of the public key"
+	}
+	private_key: {
+		[type.googleapis.com/keyspb.PrivateKey] {
+			der: ".... bytes of the private key ...."
+		}
+	}
+}
+```
+
+### Setup Environment
+
+Build the `updatetree` command if this hasn't already been done and ensure
+that it is on your `PATH`.
+
+```
+go install github.com/google/trillian/cmd/updatetree
+export PATH=${GOPATH}/bin:$PATH
+```
+
+Set environment variables to the correct log_id and metrics HTTP endpoint.
+For example:
+
+```
+LOG_ID=987654321
+METRICS_URI=http://signer-1:8091/metrics
+```
+
+## Set Log Tree To Draining State
+
+Use `updatetree` to set the log tree to a `DRAINING` state.
+
+`updatetree --tree_id=${LOG_ID} --tree-state=DRAINING`
+
+Make sure the above command succeeds. At this point the log will not
+accept new entries but there may be some that have already been
+submitted but not yet integrated.
+
+## Monitor Queue / Integration
+
+If you have monitoring dashboards showing signer mastership e.g. in
+Prometheus then this information might be easily available and you
+may already have a global view of the state of all the Trillian
+servers in the etcd cluster. For the rest of the document we will assume
+that this is not the case.
+
+The necessary information can be obtained from the raw metrics
+that the server exports. Note that it is possible for elections /
+resignations or cluster operations to change the signer responsible for a
+tree during the following process. So if you are using metrics directly
+from servers be aware that this could happen while you're watching the queue.
+
+Wait until you're sure that the log has finished integrating the
+queued leaves. This will be indicated by an incrementing count of
+signer runs for the tree, no increase in errors for the tree and zero
+leaves being processed for the tree by the signer. The following example
```

Comment:
Maybe add something about possibly needing to wait for a number of signer runs to happen, if your storage layer does not retrieve all pending leaves on a single `DequeueLeaves` operation (e.g. if it uses sharding) ? 
(or it might belong better further down).

Reply:
Done. Also added additional scary warning to the next section.